src/main/windows/mainWindow.ts

```diff
@@ -314,6 +315,7 @@ export class MainWindow extends EventEmitter {
         globalShortcut.unregisterAll();
 
         this.emit(MAIN_WINDOW_RESIZED, this.getBounds());
+        ipcMain.emit(TOGGLE_SECURE_INPUT, null, false);
```

Comment:
This is the right idea because main process is trusted and so the event here can't be forged. The web context in the renderer can still forge focus events though, so the main process `onBlur` and `onFocus` handlers should be made "authoritative". Meaning, if the window is blurred according to trusted data from the main process, any focus events from the renderer should be ignored. Events from the renderer should only be acted on if the window is already in focus.